core/src/main/java/org/apache/calcite/prepare/CalcitePrepareImpl.java

```diff
@@ -837,8 +837,16 @@ private static ColumnMetaData.AvaticaType avaticaType(JavaTypeFactory typeFactor
         : origins.get(origins.size() - 1 - offsetFromEnd);
   }
 
-  private static int getTypeOrdinal(RelDataType type) {
-    return type.getSqlTypeName().getJdbcOrdinal();
+  public static int getTypeOrdinal(RelDataType type) {
```

Comment:
formatting looks off. Also, I'd save a few lines by avoiding assigning to the variable, just `return` immediately